### DISCUSSION TITLE
using @prb/math with openzeppelin upgrades plugin

### DISCUSSION THREAD

**Discussion by yohanip** [PaulRBerg/prb-math#204] [Q&A]
Hi friends,
I tried using the contracts on a project which utilize upgrades plugin
it gave the error: _Failed to compile modified contracts for namespaced storage_
but if not using the plugin it compiled without issue

**Comment by PaulRBerg** [comment]
Hi @yohanip, thanks for your question.

To be able to assist you I will need you to share the following information:

- Development framework (Foundry, Hardhat, etc.)
- Mode of installation (git submodule, Node.js dependencies, etc.)
- Version of PRBMath
- Version of the OpenZeppelin Upgrades plugin

**Reply by yohanip** [reply]
i thank you very much for your attention, very much delighted
I am using the following combination:

- hardhat 2.17.2
- yarn 1.22.17
- PRBMath 4.0.1
- @openzeppelin/hardhat-upgrades  2.3.1

i resolved the issue by renaming the function **@ud60x18/Conversions.sol**

function convert(uint256 x) pure returns (UD60x18 result) {
to
function convertu(uint256 x) pure returns (UD60x18 result) {

seems like there is a problem with storage system when using openzeppelin's upgrade plugin, we are forced to not using **function overload**

**Reply by PaulRBerg** [reply]
Ah, I see. It looks like what you have to do is to start using [named import syntax](https://book.getfoundry.sh/tutorials/best-practices#general-contract-guidance):

- Good: `import {MyContract} from "src/MyContract.sol"` to only import `MyContract`
- Bad: `import "src/MyContract.sol"` imports everything in `MyContract.sol`
